### ISSUE TITLE
TIMELINE: Forwards to /__history__.html?0

### ISSUE THREAD

**Issue by GoogleCodeExporter** [berusch/simile-widgets-old#61]
```
I have a relatively simple timeline that loads events from an XML file. 
The timeline works perfectly fine on my local machine.

However, when I uploaded the timeline to a server, when the timeline loads,
the page is immediately forwarded to /__history__.html?0.  Sometimes I do
get a glimpse of the events from the XML file correctly displayed on the
timeline before I am forwarded to the 404 error that the /__history.html
request generates.

I am using 'http://static.simile.mit.edu/timeline/api-2.2.0/timeline-api.js'.


```

Original issue reported on code.google.com by `ericmatt...@gmail.com` on 23 Feb 2009 at 9:29

**Comment by GoogleCodeExporter** [comment]
```
In Firefox 3, it forwards as I described.  In Safari 3, the timeline appears to 
be
blank and nothing happens.
```

Original comment by `ericmatt...@gmail.com` on 23 Feb 2009 at 9:46

**Comment by GoogleCodeExporter** [comment]
```
I have also tried hosting the timeline library on the server rather than 
calling the
external javascript, and experienced the same results.
```

Original comment by `ericmatt...@gmail.com` on 23 Feb 2009 at 10:28

**Comment by GoogleCodeExporter** [comment]
```
I don't quite understand what the submitter means by "forwarded to 
/__history__.html?0"

Does he mean that the Timeline is moved to the date that is inside the history 
file?
Or something else?

Does the Timeline call include a date that the Timeline should be displayed?

A test case is needed. Otherwise can't reproduce....
```

Original comment by `larryklu...@gmail.com` on 23 Feb 2009 at 11:02

**Comment by GoogleCodeExporter** [comment]
```
The entire page is redirected to that url (ex:
http://mysite.com/timeline/__history__.html?0).
```

Original comment by `ericmatt...@gmail.com` on 23 Feb 2009 at 11:19

**Comment by GoogleCodeExporter** [comment]
```
Please supply a test case, preferably on a server. Use mock data for the 
timeline if
your production data is secret.

Thanks,

Larry
```

Original comment by `larryklu...@gmail.com` on 23 Feb 2009 at 11:44

**Comment by GoogleCodeExporter** [comment]
```
Would it matter if I am developing on an internal dev server that cannot be 
accessed
publicly?  I will try to test on a public server and provide a test case if I
experience similar problems.
```

Original comment by `ericmatt...@gmail.com` on 26 Feb 2009 at 2:53

**Comment by GoogleCodeExporter** [comment]
```
I uploaded a test case to my personal server (which is public).  I am not
experiencing the same url redirect error, so it must be something to do with
developing on a non-public server.

http://ericmatthys.com/test_timeline/test.html

I am not sure why I am still getting a blank timeline when testing through 
Safari,
however.
```

Original comment by `ericmatt...@gmail.com` on 26 Feb 2009 at 4:20

**Comment by GoogleCodeExporter** [comment]
```
I am experiencing the same issue using the example XML data. Here's a link that 
will
be available for the next few days:

http://inciweb.org/test/timeline.php 
```

Original comment by `worldcom...@gmail.com` on 27 Feb 2009 at 6:04

**Comment by GoogleCodeExporter** [comment]
```
Yes, that is the same problem I was trying to describe, so I guess it is not
necessarily a problem with a public vs internal server.
```

Original comment by `ericmatt...@gmail.com` on 2 Mar 2009 at 4:47

**Comment by GoogleCodeExporter** [comment]
```
I was able to get a temporary fix by disabling history management.  On line 
1972 in
simile-ajax-bundle.js, I set "enable" to false.  Obviously this is not ideal, 
but it
got rid of the immediate url forwarding.
```

Original comment by `ericmatt...@gmail.com` on 9 Mar 2009 at 5:46

**Comment by GoogleCodeExporter** [comment]
Original comment by `stefano.mazzocchi@gmail.com` on 25 Mar 2009 at 7:01
- Added labels: **Project-Timeline**

**Comment by GoogleCodeExporter** [comment]
```
Using WebScarab I found that the __history__.html file is never where it is 
suppose
to be.  There is a history.html file in the timeline/timeline_ajax/content 
directory
and when I copied it to the location where the code was looking for it and 
renamed
the file __history__.html.  the problem went away.  Sort of.  The timeline will 
still
not display any events from the loadXML call but I believe that may be a
configuration problem with the server.
```

Original comment by `jimmcmil...@gmail.com` on 25 Mar 2009 at 8:30

**Comment by GoogleCodeExporter** [comment]
```
Ive had similar problems. On my setup an incorrect url forwards to the default 
page,
so for every page with a timeline it is downloading the default page and 
sticking it
in an invisible iframe! Very annoying, particularly as it then also fires up the
scripts on that page. Have patched simile-ajax-bundle.js as suggested.

Some general refactoring to help out those of us that want to host locally, 
such that
urls can be set in a config object would be very desirable.
```

Original comment by `tristand...@gmail.com` on 13 Aug 2009 at 1:50

**Comment by GoogleCodeExporter** [comment]
```
No need to hack in the js file.
Just add "SimileAjax.History.enabled = false;" inside a script tag after loading
timeline-api.js and before creating the timeline.
```

Original comment by `leroyz.m...@gmx.net` on 14 Apr 2010 at 2:14

**Comment by GoogleCodeExporter** [comment]
```
Alternatively, if you still want to use the history feature (I don't like it, I 
must
say, because the implementation is rather intrusive), correct the path to the 
html
file located in the timeline folder under

timeline_2.3.0\timeline_ajax\content\history.html

like the following after loading timeline-api.js:

'SimileAjax.History.historyFile =
"/{yourScriptFolder}/timeline_{version}/timeline_ajax/content/history.html";'
```

Original comment by `leroyz.m...@gmx.net` on 14 Apr 2010 at 2:36

**Comment by GoogleCodeExporter** [comment]
```
Thanks for providing the neat solution to this problem! 
```

Original comment by `deborah....@gmail.com` on 22 Apr 2010 at 4:34

**Comment by GoogleCodeExporter** [comment]
```
Thanks. Exactly the information that I'm looking for. Thanks.
```

Original comment by `MBA....@gmail.com` on 16 Aug 2010 at 9:06

**Comment by GoogleCodeExporter** [comment]
```
Thx dude!!
```

Original comment by `seb.jacq...@gmail.com` on 6 Jun 2011 at 3:27

**Comment by GoogleCodeExporter** [comment]
```
can anyone paste code for setting the history location or 
SimileAjax.History.enabled = false code here? 

I have tried multiple ways but still it shows 404 for the URL 
"__history__.html" in http watch.

Thanks in advance.
```

Original comment by `nileshbc...@gmail.com` on 29 Jul 2011 at 8:25

**Comment by GoogleCodeExporter** [comment]
```
in the html put this

<script 
src="http://api.simile-widgets.org/timeline/2.3.1/timeline-api.js?bundle=true" 
type="text/javascript"></script>
        <script>SimileAjax.History.enabled = false;</script>
```

Original comment by `jca...@gmail.com` on 31 Aug 2011 at 3:06
